File: packages/sdk/output/typescript/src/services/linkedUsers/LinkedUsers.ts
import BaseService from '../../BaseService';

import { CreateLinkedUserDto } from './models/CreateLinkedUserDto';

import { serializeQuery } from '../../http/QuerySerializer';

export class LinkedUsersService extends BaseService {
  async linkedUsersControllerAddLinkedUser(input: CreateLinkedUserDto): Promise<any> {
    const headers: { [key: string]: string } = { 'Content-type': 'application/json' };
    const urlEndpoint = '/linked-users/create';
    const finalUrl = encodeURI(`${this.baseUrl + urlEndpoint}`);
    const response: any = await this.httpClient.post(
      finalUrl,
      input,
      {
        ...headers,
        ...this.getAuthorizationHeader(),
      },
      true,
    );
    const responseModel = response.data;
    return responseModel;
  }

  async linkedUsersControllerGetLinkedUsers(): Promise<any> {
    const urlEndpoint = '/linked-users';
    const finalUrl = encodeURI(`${this.baseUrl + urlEndpoint}`);
    const response: any = await this.httpClient.get(
      finalUrl,
      {},
      {
        ...this.getAuthorizationHeader(),
      },
      true,
    );
    const responseModel = response.data;
    return responseModel;
  }

  async linkedUsersControllerGetLinkedUser(id: string): Promise<any> {
    if (id === undefined) {
      throw new Error('The following parameter is required: id, cannot be empty or blank');
    }
    const queryParams: string[] = [];
    if (id) {
      queryParams.push(serializeQuery('form', true, 'id', id));
    }
    const urlEndpoint = '/linked-users/single';
    const finalUrl = encodeURI(`${this.baseUrl + urlEndpoint}?${queryParams.join('&')}`);
    const response: any = await this.httpClient.get(
      finalUrl,
      {},
      {
        ...this.getAuthorizationHeader(),
      },
      true,
    );
    const responseModel = response.data;
    return responseModel;
  }
}

File: packages/sdk/output/python/src/testsdk/services/linked_users.py
from urllib.parse import quote
from ..net import query_serializer
from .base import BaseService
from ..models.CreateLinkedUserDto import CreateLinkedUserDto as CreateLinkedUserDtoModel


class LinkedUsers(BaseService):
    def linked_users_controller_add_linked_user(
        self, request_input: CreateLinkedUserDtoModel
    ):
        url_endpoint = "/linked-users/create"
        headers = {"Content-type": "application/json"}
        self._add_required_headers(headers)

        final_url = self._url_prefix + url_endpoint
        res = self._http.post(final_url, headers, request_input, True)
        return res

    def linked_users_controller_get_linked_users(self):
        url_endpoint = "/linked-users"
        headers = {}
        self._add_required_headers(headers)

        final_url = self._url_prefix + url_endpoint
        res = self._http.get(final_url, headers, True)
        return res

    def linked_users_controller_get_linked_user(self, id: str):
        url_endpoint = "/linked-users/single"
        headers = {}
        query_params = []
        self._add_required_headers(headers)
        if not id:
            raise ValueError("Parameter id is required, cannot be empty or blank.")
        query_params.append(query_serializer.serialize_query("form", False, "id", id))
        final_url = self._url_prefix + url_endpoint + "?" + "&".join(query_params)
        res = self._http.get(final_url, headers, True)
        return res
